File: problems/868-binary-gap/solution.c
int binaryGap(int n) {
    int maxGap = 0;
    int lastOne = -1;
    int currentBit = 0;
    while (n > 0) {
        if (n % 2 == 1) {
            if (lastOne != -1) {
                maxGap = (currentBit - lastOne) > maxGap ? (currentBit - lastOne) : maxGap;
            }
            lastOne = currentBit;
        }
        n /= 2;
        currentBit++;
    }
    return maxGap;
}

File: problems/868-binary-gap/solution.java
class Solution {
    public int binaryGap(int n) {
        String binary = Integer.toBinaryString(n);
        int maxGap = 0;
        int lastOne = -1;
        for (int i = 0; i < binary.length(); i++) {
            if (binary.charAt(i) == '1') {
                if (lastOne != -1) {
                    maxGap = Math.max(maxGap, i - lastOne);
                }
                lastOne = i;
            }
        }
        return maxGap;
    }
}